public/pages/Dashboard/utils/utils.tsx

```diff
@@ -517,7 +522,8 @@ export const getLatestAnomalyResultsByTimeRange = async (
     );
     anomalyResults = [...anomalyResults, ...anomalies];
     from += anomalySize;
-  } while (numResults === MAX_ANOMALIES);
+    numSingleBatchResults = anomalies.length;
+  } while (numSingleBatchResults === MAX_ANOMALIES);
```

Comment:
If too many results, is it possible we will do a lot of while loops and result in timeout?

Reply:
Good point. Thanks for creating the issue #91 . Will fix it after release.